### PR TITLE
mgr/cephadm: Add example to run when debugging ssh failures

### PR DESCRIPTION
**Pull request by sebastian-philipp**
```
$ ceph orch host add foobar
Error ENOENT: Failed to connect to foobar (foobar).  Check that the host is reachable and accepts connections using the cephadm SSH key
you may want to run:
> ssh -F =(ceph cephadm get-ssh-config) -i =(ceph config-key get mgr/cephadm/ssh_identity_key) rook@foobar
$  ssh -F =(ceph cephadm get-ssh-config) -i =(ceph config-key get mgr/cephadm/ssh_identity_key) rook@foobar
ssh: Could not resolve hostname foobar: Temporary failure in name resolution
```

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1617,7 +1635,12 @@ def _run_cephadm(self, host, entity, command, args,
             # this is a misleading exception as it seems to be thrown for
             # any sort of connection failure, even those having nothing to
             # do with "host not found" (e.g., ssh key permission denied).
-            raise OrchestratorError('Failed to connect to %s (%s).  Check that the host is reachable and accepts connections using the cephadm SSH key' % (host, addr)) from e
+            user = 'rook' if self.mode == 'root' else 'cephadm'
```
Comment (liewegas) [inline src/pybind/mgr/cephadm/module.py]: s/rook/root/

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -1617,7 +1635,12 @@ def _run_cephadm(self, host, entity, command, args,
             # this is a misleading exception as it seems to be thrown for
             # any sort of connection failure, even those having nothing to
             # do with "host not found" (e.g., ssh key permission denied).
-            raise OrchestratorError('Failed to connect to %s (%s).  Check that the host is reachable and accepts connections using the cephadm SSH key' % (host, addr)) from e
+            user = 'root' if self.mode == 'root' else 'cephadm'
+            msg = f'Failed to connect to {host} ({addr}).  ' \
+                  f'Check that the host is reachable and accepts connections using the cephadm SSH key\n' \
+                  f'you may want to run: \n' \
+                  f'> ssh -F =(ceph cephadm get-ssh-config) -i =(ceph config-key get mgr/cephadm/ssh_identity_key) {user}@{host}'
```
Comment (ricardoasmarques) [inline src/pybind/mgr/cephadm/module.py]: @sebastian-philipp 
```
node1:~ # ssh -F =(ceph cephadm get-ssh-config) -i =(ceph config-key get mgr/cephadm/ssh_identity_key) root@master
-bash: syntax error near unexpected token `('
```

Comment (ricardoasmarques) [reply src/pybind/mgr/cephadm/module.py]: Nevermind, I was using `cephadm` from master

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py]: right. `get-ssh-config` was the missing piece in order to make this work.

**Comment by gusevfe** [comment]
Hello. This is zsh-specific syntax, right? I was quite surprised to see this example and running it in bash returned a syntax error. It would probably be better to at least change this to "you may want to run in zsh".

**Pull request merged**